mlabs/src/Mlabs/EfficientNFT/Marketplace.hs

```diff
@@ -2,29 +2,37 @@
 {-# LANGUAGE ImportQualifiedPost #-}
 {-# LANGUAGE TypeApplications #-}
 
-module Mlabs.EfficientNFT.Marketplace (
-  mkPolicy,
-  MintAct (MintToken, ChangePric, ChangeOwner),
-  OwnerData (..),
-  PlatformConfig (..),
-  policy,
-) where
+module Mlabs.EfficientNFT.Marketplace (mkValidator) where
 
-import Data.ByteString (ByteString)
-import Ledger qualified
-import Ledger.Crypto (PubKeyHash)
+import Ledger (
+  ScriptContext,
+  TxInInfo (txInInfoResolved),
+  TxInfo (txInfoOutputs),
+  findOwnInput,
+  scriptContextTxInfo,
+  txOutValue,
+ )
 import Ledger.Value qualified as Value
-import Plutus.V1.Ledger.Scripts qualified as Scripts
-import PlutusTx.Natural (Natural)
+import PlutusTx.Prelude
 
--- TODO
-mkValidator ::
-  TxOutRef ->
-  PubKeyHash ->
-  Natural ->
-  PlatformConfig ->
-  BuiltinData ->
-  BuiltinData ->
-  ScriptContext ->
-  Bool
-mkValidator oref authorPkh royalty platformConfig mintAct ctx = False
+-- | An escrow-like validator, that holds an NFT until sold or pulled out
+{-# INLINEABLE mkValidator #-}
+mkValidator :: BuiltinData -> ScriptContext -> Bool
+mkValidator _ ctx =
```

Comment:
for simplicity's sake let's add a Datum with nftCS to lookup the correct remint in the txInfoMinted field instead